### PR TITLE
ci: fix link checker

### LINKED ISSUE #8: CI links check always fails for the bdk/versions endpoint
See example run here: https://github.com/thunderbiscuit/book-of-bdk/actions/runs/6847755935

### PR DESCRIPTION
**Pull request by storopoli**
Adds a config .mlc_config.json to the root dir.
This enables us to filter patterns such as crates.io. I took the liberty of adding additional configs that I  am using in other repos that depend on markdown-link-check CI action.

Closes #8.

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
Awesome! Thanks for that fix.

Can I ask you what the extra config do?

**Comment by storopoli** [comment]
> Can I ask you what the extra config do?

Yes, of course, always.
I'll add some comments to the code.

File: .mlc_config.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "aliveStatusCodes": [
+    403,
+    200,
+    206
```
Comment (storopoli) [inline .mlc_config.json:5]: If we receive any of these HTTP codes, treat them as a sign that the links are alive.

File: .mlc_config.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      "urls": [
+        "https://github.com/",
+        "https://guides.github.com/",
+        "https://help.github.com/",
+        "https://docs.github.com/"
+      ],
+      "headers": {
+        "Accept-Encoding": "zstd, br, gzip, deflate"
+      }
+    }
```
Comment (storopoli) [inline .mlc_config.json:18]: Whenever checking links, to these URLs, send these header options.
This is a config that I've found in the markdown-link-checker repo
and works great so far in other link checking deployments that I have.

File: .mlc_config.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  ],
+  "retryOn429": true,
```
Comment (storopoli) [inline .mlc_config.json:20]: If we get 429s HTTP status codes,
please retry it.

File: .mlc_config.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  "ignorePatterns": [
+    {
+      "pattern": "/github/workspace/.*"
+    },
+    {
+      "pattern": "crates.io"
+    }
+  ]
```
Comment (storopoli) [inline .mlc_config.json:28]: Don't check links with theses patterns/regexes:

1. `"/github/workspace/.*"`: any internal links in the book of bdk
2. `"crates.io"`: is not friendly to markdown link checker, so we are disabling for now.

**Review by thunderbiscuit — APPROVED** [review]

**Pull request merged**